Allow to set initial probability

damianmarti opened the pull request:
- The option1 probability can be set when creating the prediction market.
- The percentage of tokens to lock can be set too.
- The liquidity provider can't sell, buy or transfer tokens.
- Removed i_virtualTradePercentage. No needed anymore, because the first tokens are locked by the prediction market owner.

damianmarti committed:
Allow to set initial probability

phipsae commented:
Nice thank you, works nice! But maybe keep the branches separate, so we keep that with the virtual trades?

damianmarti commented:
> Nice thank you, works nice! But maybe keep the branches separate, so we keep that with the virtual trades?

Great! Yes, sure, we can merge it later if we want!

phipsae commented:
When overwriting _update in the token code, the liquidity provider cannot call the function redeem anymore when the prediction got reported. I removed to _update override and added transfer and transferFrom override. Is there some case we need to thing of that is not covered yet?

phipsae committed:
remove _update overwrite and add transfer transferFrom override

damianmarti commented:
> When overwriting _update in the token code, the liquidity provider cannot call the function redeem anymore when the prediction got reported. I removed to _update override and added transfer and transferFrom override. Is there some case we need to thing of that is not covered yet?

Oh, you are right!! Sorry for not testing that! The burn function uses an update too :-/

phipsae commented:
No worries :) 
I also saw that the owner can always mint tokens when he wants, that also an issue in general, since it dilutes the holder of tokens. 
Burn true, but what is the issue? If the owner burns token that has no disadvantage for the users only for the owner? Or do I see something wrong?

damianmarti commented:
> No worries :) I also saw that the owner can always mint tokens when he wants, that also an issue in general, since it dilutes the holder of tokens. Burn true, but what is the issue? If the owner burns token that has no disadvantage for the users only for the owner? Or do I see something wrong?

no, no, it's ok.

phipsae commented:
> > No worries :) I also saw that the owner can always mint tokens when he wants, that also an issue in general, since it dilutes the holder of tokens. Burn true, but what is the issue? If the owner burns token that has no disadvantage for the users only for the owner? Or do I see something wrong?
> 
> no, no, it's ok.

Also the mint? But can't the lp always mint token out of nowhere?

phipsae commented:
> > No worries :) I also saw that the owner can always mint tokens when he wants, that also an issue in general, since it dilutes the holder of tokens. Burn true, but what is the issue? If the owner burns token that has no disadvantage for the users only for the owner? Or do I see something wrong?
> 
> no, no, it's ok.

True because only the prediction market can mint/burn
